internal/controller/deployment/controller_finalize.go

```diff
@@ -87,3 +129,48 @@ func (r *Reconciler) finalize(ctx context.Context, old, deployment *choreov1.Dep
 	}
 	return ctrl.Result{}, nil
 }
+
+func (r *Reconciler) cleanupEndpoints(ctx context.Context, deployment *choreov1.Deployment) error {
+	logger := log.FromContext(ctx).WithValues("deployment", deployment.Name)
+	logger.Info("Cleaning up the endpoints associated with the deployment")
+
+	// List all the endpoints associated with the deployment
+	endpointList := &choreov1.EndpointList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(deployment.Namespace),
+		client.MatchingLabels{
+			labels.LabelKeyDeploymentName: deployment.Name,
```

Comment:
Shall we add the full hierarchy labels here? Otherwise, it can list endpoints belonging to another component or project.

Reply:
Added the full hierarchy here.